nix-mode.el

```diff
@@ -622,7 +724,17 @@ The hook `nix-mode-hook' is run when Nix mode is started.
   (setq-local parse-sexp-lookup-properties t)
 
   ;; Automatic indentation [C-j]
-  (setq-local indent-line-function nix-indent-function)
+  (smie-setup nix-smie-grammar 'nix-smie-rules
+              :forward-token 'nix-smie--forward-token
+              :backward-token 'nix-smie--backward-token)
+  (setq-local smie-indent-basic 2)
+  (setq-local indent-line-function 'smie-indent-line)
+  (ignore-errors
+    (setf (car (memq 'smie-indent-exps smie-indent-functions))
```

Comment:
Why not just ```(setf 'smie-indent-exps 'nix-smie--indent-exps)```?

Reply:
Also we should make sure this is local? I hope it doesn't carry over to other smie-based modes.

Reply:
Good catch, this does indeed carry over to all other buffers. I've made the variables buffer-local and use `fset` to change the value in the function cell.